volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -843,25 +855,55 @@ def get_subdirs(self) -> interfaces.objects.ObjectInterface:
         dentry_type_name = self.get_symbol_table_name() + constants.BANG + "dentry"
         yield from list_head_member.to_list(dentry_type_name, walk_member)
 
+    def get_inode(self) -> interfaces.objects.ObjectInterface:
+        """Returns the inode associated with this dentry"""
+
+        inode_ptr = self.d_inode
+        if not (inode_ptr and inode_ptr.is_readable() and inode_ptr.is_valid()):
+            return None
+
+        return inode_ptr.dereference()
+
 
 class struct_file(objects.StructType):
     def get_dentry(self) -> interfaces.objects.ObjectInterface:
-        if self.has_member("f_dentry"):
-            return self.f_dentry
-        elif self.has_member("f_path"):
+        """Returns a pointer to the dentry associated with this file"""
+        if self.has_member("f_path"):
```

Comment:
Same as in other code I just reviewed, will the change in order have an impact on the results returned and if so, do we need to bump a version number somewhere so people know?

Reply:
I moved the `if` matching the most recent kernels to be first, which it's likely to be the most often case now and in the future.

Reply:
re: version.. there's no version for this

Reply:
Ok, cool, just wanted to check.  5:)